### PR TITLE
4654 quick filter dashboard

### PR DESCRIPTION
**Pull request by MV88**
## Description

Add support for quick filtering in Table Widgets

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature

## Issue

**What is the current behavior?**

#4654 

**What is the new behavior?**

Actually you can add a quick filter for table widgets.
4654 quick filter table widget

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
@offtherailz 
the getWidgetFilterRenderers.js has old require syntax because i had to change too many files for this kind of migration

### CODE REVIEW

File: web/client/plugins/widgets/getWidgetFilterRenderers.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+const { connect } = require('react-redux');
+const { createSelector } = require('reselect');
+
+const { getAttributeFields } = require('../../utils/FeatureGridUtils');
+const { getFilterRenderer } = require('../../components/data/featuregrid/filterRenderers');
+const { getWidgetAttributeFilter } = require('../../selectors/widgets');
+
+
+const getWidgetFilterRenderers = createSelector(
```
Comment (offtherailz) [inline web/client/plugins/widgets/getWidgetFilterRenderers.js]: You have all the information as properties for the widget, no need to connect anything.
I should create a withFilters enhancer that parses the describefeauretype and provides the proper filterRenderer at the contained Component. If you want to cache, you can use withPropsOnChange.

Here you're using a selector that is not state selector but a function... it's very weird.

File: web/client/components/widgets/widget/__tests__/TableWidget-test.jsx [hunk 2]
```diff
@@ -69,7 +70,15 @@ describe('TableWidget component', () => {
...
-        ReactDOM.render(<TableWidget describeFeatureType={describePois} features={[]} />, document.getElementById("container"));
+        const store = {
+            dispatch: () => {},
+            subscribe: () => {
+                return () => {};
+            },
+            unsubscribe: () => {},
+            getState: () => ({})
+        };
+        ReactDOM.render(<Provider store={store}><TableWidget describeFeatureType={describePois} features={[]} /></Provider>, document.getElementById("container"));
```
Comment (offtherailz) [inline web/client/components/widgets/widget/__tests__/TableWidget-test.jsx]: This test should be restored as well as the table widget depends only from props

File: web/client/components/widgets/widget/TableWidget.jsx [hunk 3]
```diff
@@ -67,6 +68,8 @@ module.exports = ({
                 columnSettings={columnSettings}
                 pageEvents={pageEvents}
                 virtualScroll={virtualScroll}
+                enableColumnFilters
+                filterRenderers={getWidgetFilterRenderers(describeFeatureType, id)}
```
Comment (offtherailz) [inline web/client/components/widgets/widget/TableWidget.jsx]: create an enhancer that adds filterRenderers prop, you have all the information you need to create filter renderers from prop, no need to do strange things (also attributeFilter should be available at this scope)

File: web/client/plugins/widgets/getWidgetFilterRenderers.js (hunk 1 above)
Comment (MV88) [reply web/client/plugins/widgets/getWidgetFilterRenderers.js]: ok i'll change it, but it's quite funny how  you call "weird" [your](https://github.com/geosolutions-it/MapStore2/blame/c33eadbc5f900b2b26ca9f18a888ff2dbd3272f9/web/client/plugins/featuregrid/panels/index.jsx#L137) creations :)

File: web/client/components/widgets/widget/__tests__/TableWidget-test.jsx (hunk 2 above)
Comment (MV88) [reply web/client/components/widgets/widget/__tests__/TableWidget-test.jsx]: sure

**Pull request merged**